modules/freeWheelAdserverVideo.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * This module adds Freewheel support for Video to Prebid.
+ */
+
+import { registerVideoSupport } from '../src/adServerManager';
+import { auctionManager } from '../src/auctionManager';
+import { groupBy, deepAccess } from '../src/utils';
+import { config } from '../src/config';
+// import { ADPOD, initAdpodHooks } from './adpod';
+
+const ADPOD = 'adpod'; // remove later when adpod module is merged; use above commented import instead
+const adPodTargetingKey = 'hb_price_industry_duration';
+
+/**
+ * This function returns targeting keyvalue pairs for freewheel adserver module
+ * @param {Object} options
+ * @param {Array[string]} codes
+ * @returns targeting kvs for adUnitCodes
+ */
+export default function getTargeting({codes} = {}) {
+  codes = codes || [];
+  const adPodAdUnits = getAdPodAdUnits(codes);
+  const bidsReceived = auctionManager.getBidsReceived();
+  const competiveExclusionEnabled = config.getConfig('adpod.brandCategoryExclusion');
+
+  let bids = getBidsForAdpod(bidsReceived, adPodAdUnits);
+  bids = (competiveExclusionEnabled) ? getExclusiveBids(bids) : bids;
+  bids.sort(compare);
+
+  let targeting = {};
+  adPodAdUnits.forEach((adUnit) => {
+    let adPodTargeting = [];
+    let adPodDurationSeconds = deepAccess(adUnit, 'mediaTypes.video.adPodDurationSec');
+
+    bids
+      .filter((bid) => bid.adUnitCode === adUnit.code)
+      .forEach((bid, index, arr) => {
+        if (bid.video.durationSeconds <= adPodDurationSeconds) {
+          adPodTargeting.push({
+            [adPodTargetingKey]: bid.adserverTargeting[adPodTargetingKey]
+          });
+          adPodDurationSeconds -= bid.video.durationSeconds;
+        }
+        if (index === arr.length - 1 && adPodTargeting.length > 0) {
+          adPodTargeting.push({
+            'hb_uuid': bid.adserverTargeting.hb_uuid
+          });
+        }
+      });
+    targeting[adUnit.code] = adPodTargeting;
+  });
+  return targeting;
+}
+
+/**
+ * This function returns the adunit of mediaType adpod
+ * @param {Array} codes adUnitCodes
+ * @returns {Array[Object]} adunits of mediaType adpod
+ */
+function getAdPodAdUnits(codes) {
+  return $$PREBID_GLOBAL$$.adUnits
```

Comment:
What if someone uses the `adUnits` field when executing the `$$PREBID_GLOBAL$$.makeBidRequests()` - could that cause issues here (like if they were refreshing a particular adunit instead of everything that may have originally been set)?

Reply:
I was under the impression that we modify adUnits inside requestBids. Hence used the global. Now modified to use auctionManager.getAdUnits